Update docs to include more information about kerchunk (#469)
pangeo-forge/pangeo-forge-recipes: docs/pangeo_forge_recipes/tutorials/index.md

These tutorials all use the {class}`pangeo_forge_recipes.recipes.HDFReferenceRecipe` class:

> Unlike the standard `XarrayZarrRecipe` recipe creation method, these reference recipes utilize [kerchunk](https://fsspec.github.io/kerchunk/) to create a ReferenceRecipe. The difference between these two methods is that the standard XarrayZarrRecipe creates a copy of the original dataset in the Zarr format, while the kerchunk-based ReferenceRecipe does not copy the data and instead creates a kerchunk mapping, which allows legacy formats such as (NetCDF, GRIB2, TIFF, FITS etc.) to be read as if they were Zarr stores. More details about how kerchunk works can be found in the [kerchunk docs](https://fsspec.github.io/kerchunk/detail.html). Kerchunk can be used independently outside of Pangeo-Forge. In these examples, Pangeo-Forge acts as a runner for kerchunk. Currently only HDF/NetCDF Reference Recipes are supported by Pangeo-Forge.

Reviewer: ```suggestion
Unlike the standard `XarrayZarrRecipe` recipe creation method, these reference recipes utilize [kerchunk](https://fsspec.github.io/kerchunk/) to create a `ReferenceRecipe`. The difference between these two methods is that the standard `XarrayZarrRecipe` creates a copy of the original dataset in the Zarr format, while the kerchunk-based `ReferenceRecipe` does not copy the data and instead creates a kerchunk mapping, which allows legacy formats such as (NetCDF, GRIB2, TIFF, FITS etc.) to be read as if they were Zarr stores. More details about how kerchunk works can be found in the [kerchunk docs](https://fsspec.github.io/kerchunk/detail.html). Kerchunk can be used independently outside of Pangeo-Forge. In these examples, Pangeo-Forge acts as a pipeline for generating kerchunk references. Currently only HDF/NetCDF Reference Recipes are supported by Pangeo-Forge.
```

Are you sure that GRIB is not supported?

Reviewer 2: https://github.com/pangeo-forge/pangeo-forge-recipes/pull/438 ?

Reviewer 2: "as if they were Zarr stores."

Something like:
"as if they were Zarr stores. This means that the users get the parallel chunk-wise access of a modern cloud-native library while reading the original data files in situ, in their original format".

Reviewer: 👍  to "dataset" over "store". 

We have no glossary. That would be useful though.

Author: Good call, updated!

Author: @rabernat , as far as I know, a grib2 reference recipe isn't supported yet. I chatted with Charles a bit ago about opening up a PR for a more generalized reference recipe class instead of reference recipe classes for each data type. The current WIP is here: https://github.com/pangeo-forge/pangeo-forge-recipes/pull/452/files.

> Unlike the standard `XarrayZarrRecipe` recipe creation method, these reference recipes utilize [kerchunk][(https://fsspec.github.io/kerchunk/) to create a ReferenceRecipe. The difference between these two methods is that the standard XarrayZarrRecipe creates a copy of the original dataset in the Zarr format, while the kerchunk-based ReferenceRecipe does not copy the data and instead creates a kerchunk mapping, which allows legacy formats such as (NetCDF, GRIB2, TIFF, FITS etc.) to be read as if they were Zarr stores. More details about how kerchunk works can be found in the [kerchunk docs](https://fsspec.github.io/kerchunk/detail.html). Kerchunk can be used independently outside of Pangeo-Forge. In these examples, Pangeo-Forge acts as a runner for kerchunk. Currently only HDF/NetCDF Reference Recipes are supported by Pangeo-Forge.

Reviewer 3: ```suggestion
Unlike the standard `XarrayZarrRecipe` recipe creation method, these reference recipes utilize [kerchunk](https://fsspec.github.io/kerchunk/) to create a ReferenceRecipe. The difference between these two methods is that the standard XarrayZarrRecipe creates a copy of the original dataset in the Zarr format, while the kerchunk-based ReferenceRecipe does not copy the data and instead creates a kerchunk mapping, which allows legacy formats such as (NetCDF, GRIB2, TIFF, FITS etc.) to be read as if they were Zarr stores. More details about how kerchunk works can be found in the [kerchunk docs](https://fsspec.github.io/kerchunk/detail.html). Kerchunk can be used independently outside of Pangeo-Forge. In these examples, Pangeo-Forge acts as a runner for kerchunk. Currently only HDF/NetCDF Reference Recipes are supported by Pangeo-Forge.
```

### When to use Reference Recipes

When choosing which recipe object you want to use, it is important to consider the end use cases of your dataset as both recipe classes have pros and cons.

> In general, ReferenceRecipes store and copy much less data overall. However, modification of the underlying data is more limited, for example, the chunking schema of a dataset can not be modified with kerchunk, so you are dependent on the initial chunk schema of the dataset. If you need to optimize your datasets chunking schema for space or time, the standard `XarrayZarrRecipe` might be a better bet.

Reviewer: ```suggestion
In general, ReferenceRecipes store and copy much less data overall. However, modification of the underlying data is more limited, for example, the chunking schema of a dataset can not be modified with kerchunk, so you are dependent on the initial chunk schema of the dataset. If you need to optimize your datasets chunking schema for space or time, the standard `XarrayZarrRecipe` might be a better bet.

If the original data are not already in the cloud, there is little benefit to using a `ReferenceRecipe`, because the access time for the data will be very slow.
```

Reviewer 2: My version:

> ReferenceRecipes do not copy the original data but instead  produce and store reference sets which are much smaller. They also don'e usually need to download/read all the bytes of the original files. However, modification of the underlying data is nor possible, for example, the chunking schema of a dataset cannot be modified with kerchunk, so you are limited to the initial chunk schema of the dataset. If you need to optimize your datasets chunking schema for space or time, the standard `XarrayZarrRecipe` is the only option. Changes in the metadata (attributes, encoding, etc.) can however be applied.

For the new sentence by @rabernat , I would include this further up, something about accessing the data "in situ". I would argue that it might still be useful for data stored in some on-prem datacentre, semi-cloud. Maybe we can link to some text elsewhere in pangeo-forge's docs which says that one of the points of pangeo-forge is to bring datasets from slow servers into the cloud for rapid public access.

Author: @martindurant, where would you prefer the bit about data access time to go. Should it be at the top of the `### When to use Reference Recipes` section or above?

Also, what do you think of this rewording to the sentence added by Ryan, des this address your comment or would something else be better?

```If the original data are not already in the cloud, there may be little benefit to using a `ReferenceRecipe`, because the access time for the data could be very slow. However, it may still be useful for data stored at on-premises data centers, with semi-cloud like access.```
